src/custom_flags/rustfix.rs

```diff
@@ -53,115 +63,51 @@ impl Flag for RustfixMode {
-                    env_vars: config.collect(|r| r.env_vars.iter().cloned()),
-                    normalize_stderr: vec![],
-                    normalize_stdout: vec![],
-                    error_in_other_files: vec![],
-                    error_matches: vec![],
-                    require_annotations_for_level: Default::default(),
-                    diagnostic_code_prefix: OptWithLine::new(String::new(), Span::default()),
-                    custom: config.comments().flat_map(|r| r.custom.clone()).collect(),
-                    exit_status: OptWithLine::new(0, Span::default()),
-                    require_annotations: OptWithLine::default(),
-                },
-            ))
-            .collect(),
-        };
-        let config = TestConfig {
-            config: config.config.clone(),
-            comments: &rustfix_comments,
-            aux_dir: config.aux_dir,
-            status: config.status,
+                stderr: output.stderr.clone(),
+                stdout: output.stdout.clone(),
+            })?
+        } else {
+            Vec::new()
         };
 
-        let run = fixed_code.is_some();
-        let mut errors = vec![];
-        let rustfix_path = config.check_output(
-            // Always check for `.fixed` files, even if there were reasons not to run rustfix.
-            // We don't want to leave around stray `.fixed` files
-            fixed_code.unwrap_or_default().as_bytes(),
-            &mut errors,
-            "fixed",
-        );
-        // picking the crate name from the file name is problematic when `.revision_name` is inserted,
-        // so we compute it here before replacing the path.
-        let crate_name = config
-            .status
-            .path()
-            .file_stem()
-            .unwrap()
-            .to_str()
-            .unwrap()
-            .replace('-', "_");
-        let config = TestConfig {
-            config: config.config,
-            comments: &rustfix_comments,
-            aux_dir: config.aux_dir,
-            status: &config.status.for_path(&rustfix_path),
+        let mut errors = Vec::new();
+        let fixed_paths = match fixed.as_slice() {
+            [] => Vec::new(),
+            [single] => {
+                vec![config.check_output(single.as_bytes(), &mut errors, "fixed")]
+            }
+            _ => {
+                config.check_output(&[], &mut errors, "fixed");
+                fixed
+                    .iter()
+                    .enumerate()
+                    .map(|(i, fix)| {
+                        config.check_output(
+                            fix.as_bytes(),
+                            &mut errors,
+                            &format!("{}.fixed", i + 1),
+                        )
+                    })
+                    .collect()
+            }
         };
+
+        if fixed.len() != 1 {
+            // Remove an unused .fixed file
+            config.check_output(&[], &mut errors, "fixed");
```

Comment:
this will fail to remove `.1.fixed` files. Maybe we need to use the `glob` crate to look for specific patterns of files to remove now

Reply:
Yeah I wasn't sure if I should do a `read_dir` or something along the lines to catch the other ones, it would be quadratic if we're checking it for every `.fixed` file

Perhaps we could check `.1.fixed`, `.2.fixed` etc until we stop finding them

Reply:
Yea... let's leave it as is. Maybe i'll record all files in the future and check them once against the actual file list